src/main/webapp/script.js

```diff
@@ -168,6 +170,60 @@ function loadProfile() {
   });
 }
 
+//matches.html functions
+function displayMatches() {
+  let id = getCurrentUserId();
+  if (id === null) {
+      return;
+  }
+  fetch('/matches?id=' + id).then(response => response.json()).then((matches) => {
+    console.log(matches);
+    for (let i = 0; i < matches.length; i++) {
+      createCardElement(matches[i]);
+    }
+  });
+}
+
+function createCardElement(userID) {
```

Comment:
It seems that the `createCardElement` function is both creating the card element and adding it to the match container, but the naming just implies that an element is created.

 It may help to split up this functionality here so that this function returns that card element being created, and then the `displayMatches` function can add that element to the matches container to be displayed.